### PR TITLE
[DO NOT MERGE] added SimpleMinHeap

### PR DESCRIPTION
**Pull request by masters3d**
waiting for https://github.com/exercism/problem-specifications/pull/1128

Missing Readme.md

### CODE REVIEW

**Comment by robtimp** [comment]
Looks good so far!

File: exercises/simple-min-heap/Sources/SimpleMinHeap/SimpleMinHeap.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            let leftChildIndex = (2 * index) + 1
+            guard leftChildIndex < array.count, index != array.count - 1
+                else { break } // No more children or current item is the last in the array
+            let rightChildIndex = leftChildIndex + 1
+
+            // There is a right child with the name parent node
+            if(rightChildIndex < array.count &&  ((rightChildIndex - 1)/2) == index ) {
+                let left = array[leftChildIndex]
+                let right = array[rightChildIndex]
+                let minChildIndex = left < right ? leftChildIndex : rightChildIndex
```
Comment (masters3d) [inline exercises/simple-min-heap/Sources/SimpleMinHeap/SimpleMinHeap.swift]: If left == right, which side should I choose? Does it matter? This impl chooses right side.

File: exercises/simple-min-heap/Sources/SimpleMinHeap/SimpleMinHeap.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            array.swapAt(elementIndex, parentIndex)
+            elementIndex = parentIndex
+            parentIndex = (elementIndex - 1 ) / 2
+            if (elementIndex == 0) {
+                break // We reached the root node
+            }
+        }
+    }
+    // Convinience function that takes in a array converts it into a heap
+    mutating func heapify(_ input: [T]) {
```
Comment (masters3d) [inline exercises/simple-min-heap/Sources/SimpleMinHeap/SimpleMinHeap.swift]: Implement a better heapifi. Not sure how to test though. Ideas https://github.com/williamfiset/data-structures/blob/master/PriorityQueue/PQueue.java

**Comment by masters3d** [comment]
Too Specific Seems like: https://github.com/exercism/problem-specifications/issues/761

**Pull request closed without merging**